### ISSUE TITLE
Ship docker-compose or rancher-compose with rancher-os

### ISSUE THREAD

**Issue by Nicofuma** [rancherio/os#323]
docker-compose (or rancher-compose) is great often used when you have to deployed services involving many containers. It would be nice to have it natively supported by rancher-os

**Comment by imikushin** [comment]
rancher-compose is what powers services on RancherOS. Perhaps we should also expose it as a command line tool.

**Comment by Nicofuma** [comment]
it make sense to me to have it under a `ros` sub-command

**Comment by pwFoo** [comment]
Will rancher-compose as sub-command of ros also usable for user-docker or only with system-docker?

**Comment by michaellopez** [comment]
+1

**Comment by ibuildthecloud** [comment]
The problem with shipping docker-compose is that it will increase the distribution size by about 9 mb.  Rancher-compose could be included as its go, but it seemed self serving to bundle it.

I have this other tool I've fooled with for awhile but have not released called "wonka".  Essentially it makes it look like you are running a local CLI but in fact it's wrapped in a docker container.  I think this is the correct route to bundling more great tools like docker compose.

**Comment by pwFoo** [comment]
> Essentially it makes it look like you are running a local CLI but in fact it's wrapped in a docker container.

Would be a great feature / tool!!!

**Comment by michaellopez** [comment]
@ibuildthecloud I look forward to it

**Comment by MetalArend** [comment]
is there a preferenced way or documentation on how to add docker-compose yourself?

**Comment by MetalArend** [comment]
my bad, just found https://github.com/rancher/os/issues/128

**Comment by zot24** [comment]
@ibuildthecloud will be good have more info about how to use `wonka` it's really hard to start working with RancherOS and make it your default OS to work daily

**Comment by SoreGums** [comment]
Using a custom container and docker network, I can achieve the same thing as docker-compose. would be nice though from an automation/scripting point of view to simply just have docker-compose :)

**Comment by scls19fr** [comment]
I was also looking for `docker-compose`.

I did

```
sudo ros console switch debian
sudo apt-get install wget
wget https://github.com/docker/compose/releases/download/1.16.1/run.sh 
chmod +x run.sh
mkdir ~/bin
mv run.sh ~/bin/docker-compose
export PATH=$PATH:~/bin
```

I put a `docker-compose.yml` in a directory and run it using:

```
docker-compose up -d
```

But having docker-compose directly could be a nice feature

**Comment by transifex3** [comment]
Not any official comments about this yet?

**Comment by NetRoller** [comment]
Another simple way to do this an keep it on each reboot, add this to your cloud-config.yml:
```
runcmd:
    - wget https://github.com/docker/compose/releases/download/1.21.2/docker-compose-$(uname -s)-$(uname -m) -O /usr/bin/docker-compose
    - chmod +x /usr/bin/docker-compose
```

**Comment by pwFoo** [comment]
@netroller
So it will BE downloaded every reboot? Maybe it's better to download it to a persistent directory and symlink if needed?

Sometimes I use docker-compose inside of a docker container (bash script or alias) 😁

**Comment by uuf6429** [comment]
This issue is open since 3 years. Is anyone actually using RancherOS, and if so, how is it being used if there is no way to orchestrate a bunch of services? The `cloud-config.yml` has a grave issue that it does not allow you to preload data before starting the services, with the worst side-effect being that non-existent files bind-mounted to volumes will be created as directories, making the affected container completely unusable (restarting the container with a directory mount replaced by a correct file mount will just show an error).

I've been trying to use ROS since several weeks now and whichever way I go, something unexpected and completely blocking stops any progress; after giving up trying to use cloud-config->rancher->services, I got stuck by the lack of a *-compose tool. The next option I'm trying is to use `docker-compose` via a docker image, which feels like a ridiculous hack.

**Comment by kuhnroyal** [comment]
I am installing docker-compose manually for a 2 years now via a simple script.
```
write_files:
  - path: /etc/setup.sh
    permissions: "0755"
    owner: root
    content: |
      #!/bin/bash
      if [ $(/usr/bin/id -u) -ne 0 ]; then
          echo "Not running as root"
          exit
      fi

      mkdir -p /opt/bin
      COMPOSE_VERSION=`curl -s https://api.github.com/repos/docker/compose/releases/latest | grep 'tag_name' | cut -d\" -f4`
      sh -c "curl -L https://github.com/docker/compose/releases/download/${COMPOSE_VERSION}/docker-compose-`uname -s`-`uname -m` > /opt/bin/docker-compose"
      chmod +x /opt/bin/docker-compose
      sh -c "curl -L https://raw.githubusercontent.com/docker/compose/${COMPOSE_VERSION}/contrib/completion/bash/docker-compose > /etc/bash_completion.d/docker-compose"
```

Would be nice if there is something build in.

**Comment by scls19fr** [comment]
An official answer here could be to show how to install [`rancher-compose`](https://github.com/rancher/rancher-compose) in RancherOS and to show what it can be used in a quite similar manner than `docker-compose`

**Comment by niusmallnan** [comment]
@scls19fr `ros service` command means `rancher-compose`. https://rancher.com/docs/os/v1.x/en/installation/system-services/custom-system-services/

Also you can download the `rancher-compose` CLI, https://github.com/rancher/rancher-compose/releases
